sections/quick-order-list.liquid

```diff
@@ -128,7 +128,18 @@
         {% comment %} Populated by JS {% endcomment %}
       </span>
     {%- else -%}
-      <div class="quick-order-list__total gradient" id="quick-order-list-total">
+      {% liquid
```

Comment:
This makes sense for now but ideally we want to check for this on the product level so we dont have to loop.

We can also add `break` when it becomes true and can stop looping 
https://shopify.dev/docs/api/liquid/tags/break

Another idea is:

What I suggest you do is
In the row file you can check if there has_popover https://github.com/Shopify/dawn/blob/main/snippets/quick-order-list-row.liquid#L196 and add a class

In the quick order list file, you can use the `:has` property https://developer.mozilla.org/en-US/docs/Web/CSS/:has in CSS to see if it includes a row with popover.

This would avoid the loop. It's not available in all browsers but since this doesnt break the experience (just changes the spacing), I think it's a fair trade off

Ill let the 2nd reviewer weigh in as well

Ideally we dont want to loop through all of the variants to find out if the product has qty. This gets re-rendered every time the section re-renders and is re-calculated

Reply:
@sofiamatulis I looked again into this issue and I don't think we need to avoid iteration because we already do the same iteration [here](https://github.com/Shopify/dawn/pull/2923/files#diff-b1e86f8491d31b61e045b54da8bb8ca6d0cc4dbb55f3560aace18423de6e176aR99). So, I just moved my logic to the existing `for` loop and removed the one I added before. 

What do you think?